js/cpu6502.ts

```diff
@@ -1173,36 +1196,82 @@ export default class CPU6502 {
         this.pc = pc;
     }
 
-    public dumpPC(pc: word | undefined, symbols: symbols) {
-        if (pc === undefined) {
-            pc = this.pc;
+    public getDebugInfo(): DebugInfo {
+        const b = this.readByteDebug(this.pc);
+        const op = this.opary[b];
+        const size = sizes[op.mode];
+        const cmd = new Array(size);
+        cmd[0] = b;
+        for (let idx = 1; idx < size; idx++) {
+            cmd[idx] = this.readByteDebug(this.pc + idx);
+        }
+
+        return [
+            this.pc,
+            this.ar,
+            this.xr,
+            this.yr,
+            this.sr,
+            this.sp,
+            cmd
+        ];
+    }
+
+    public printDebugInfo(info: DebugInfo, symbols: symbols) {
+        let symbol = '          ';
+        if (symbols && symbols[info[0]]) {
+            symbol = symbols[info[0]];
+            symbol +=  '          '.substring(symbol.length);
         }
-        const b = this.readByteDebug(pc),
-            op = this.opary[b],
-            size = sizes[op.mode];
-        let result = toHex(pc, 4) + '- ';
+
+        return [
+            toHex(info[0], 4),
+            '- ', symbol,
+            ' A=', toHex(info[1]),
+            ' X=', toHex(info[2]),
+            ' Y=', toHex(info[3]),
+            ' P=', toHex(info[4]),
+            ' S=', toHex(info[5]),
+            ' ',
+            ((info[4] & flags.N) ? 'N' : '-'),
+            ((info[4] & flags.V) ? 'V' : '-'),
+            '-',
+            ((info[4] & flags.B) ? 'B' : '-'),
+            ((info[4] & flags.D) ? 'D' : '-'),
+            ((info[4] & flags.I) ? 'I' : '-'),
+            ((info[4] & flags.Z) ? 'Z' : '-'),
+            ((info[4] & flags.C) ? 'C' : '-'),
+            ' ',
+            this.dumpRawOp(info[6]),
+            ' ',
+            this.dumpOp(info[0], info[6], symbols)
+        ].join('');
+    }
+
+    public dumpPC(_pc: word, symbols: symbols) {
```

Comment:
This seems like it could be made easier to use. Maybe something like:
```
debugger.setSymbols(symbols: symbols);
debugger.dumpPC();
debugger.dumpPC(0x8000);
...
```
It just seems odd to have the first argument be optional, but more important than the second.